crypto/s2n_certificate.c

```diff
@@ -275,14 +273,13 @@ int s2n_cert_chain_and_key_load_cns(struct s2n_cert_chain_and_key *chain_and_key
          * direct ASCII equivalent. Any non ASCII bytes in the string will fail later when we
          * actually compare hostnames.
          */
-        DEFER_CLEANUP(unsigned char *utf8_str, OPENSSL_free_pointer);
+        DEFER_CLEANUP(unsigned char *utf8_str, s2n_crypto_free);
         const int utf8_out_len = ASN1_STRING_to_UTF8(&utf8_str, asn1_str);
         if (utf8_out_len < 0) {
             /* On failure, ASN1_STRING_to_UTF8 does not allocate any memory */
             continue;
         } else if (utf8_out_len == 0) {
             /* We still need to free memory here see https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-7521 */
-            OPENSSL_free(utf8_str);
```

Comment:
why are we removing this?

Reply:
Because it will be freed up using `s2n_crypto_free`

Reply:
We need to understand why this double-free was added and if it still holds true. It seems to me that  line 285 was not being called at all.

Reply:
https://github.com/openssl/openssl/blob/1c0eede9827b0962f1d752fa4ab5d436fa039da4/crypto/asn1/a_strex.c#L624-L625